development/auto-changelog.js

```diff
@@ -69,6 +69,37 @@ async function main() {
   const changelog = await fs.readFile(changelogFilename, { encoding: 'utf8' });
   const changelogLines = changelog.split('\n');
 
+  const prNumbersWithChangelogEntries = [];
```

Comment:
This was moved earlier in the script to allow the early exit on line 99, to ensure we don't add the 'Uncategorized' header to the "Unreleased" section if there are no changes.

As a bonus, it now avoids creating a new release header if there are no commits as well! Though that shouldn't happen generally.